src/app/wallets/get-pending-onchain-balance-for-wallet.ts

```diff
@@ -1,15 +1,35 @@
-import { CouldNotFindError } from "@domain/errors"
+import {
+  CouldNotFindError,
+  MultipleCurrenciesForSingleCurrencyOperationError,
+} from "@domain/errors"
 
 import { WalletOnChainPendingReceiveRepository } from "@services/mongoose"
 import { IncomingOnChainTxHandler } from "@domain/bitcoin/onchain/incoming-tx-handler"
+import { WalletCurrency, ZERO_CENTS, ZERO_SATS } from "@domain/shared"
 
 export const getPendingOnChainBalanceForWallets = async <S extends WalletCurrency>(
   wallets: Wallet[],
 ): Promise<{ [key: WalletId]: PaymentAmount<S> } | ApplicationError> => {
   const pendingIncoming = await WalletOnChainPendingReceiveRepository().listByWalletIds({
     walletIds: wallets.map((wallet) => wallet.id),
   })
-  if (pendingIncoming instanceof CouldNotFindError) return {}
+
+  if (pendingIncoming instanceof CouldNotFindError) {
+    const settlementCurrencies = new Set(wallets.map((wallet) => wallet.currency))
```

Comment:
Shouldn't this validation happen before we even call `listByWalletIds`?

Reply:
Also if `Wallet` was parameterised I guess this would be caught by the compiler right?